io/io/src/TKey.cxx

```diff
@@ -592,6 +598,8 @@ void TKey::FillBuffer(char *&buffer)
    tobuf(buffer, version);
 
    tobuf(buffer, fObjlen);
+   if (GetFile() && GetFile()->TestBit(TFile::kReproducible))
```

Comment:
humm ... since the flag is 'just' a bit, technically it can be turn on and off during the lifetime of the file.  What about using:
```
if (GetFile() && GetFile()->TestBit(TFile::kReproducible))
    TDatime(1).FillBuffer(buffer);
else
    fDatime.FillBuffer(buffer);
```
(and also skip/remove in the change in TKey::Create.

Reply:
I used this solution mostly everywhere. Only for TKey it does not work, while TFile is not set for temporary TKey objects